add Dylibso WebAssembly Observe SDK (#2073)
DataDog/integrations-extras: webassembly-observe-sdk/README.md

# Dylibso WebAssembly Observe SDK

> ## Overview

Reviewer: The overview should include: 

1. What does WebAssemble do in 1-2 sentences?
2. What will customers get from this integration?
3. What types of data (and the value of this data)?

This integration provides function traces from WebAssembly code running in your application. Gain insights into WebAssembly code performance and behavior
including function call duration, execution tracing, and memory allocation.

Since WebAssembly code is executed in a secure & constrained environment, traditional techniques to monitor code do not work. Our specialized observability stack

> makes it easy to continuously monitor WASM modules at the same level you expect of your other applications.

Reviewer 2: ```suggestion
allows you to continuously monitor WASM modules at the same level you expect of your other applications.
```

Datadog customers can use our open source SDKs and Adapters to emit full traces from their WASM programs.

Please see the [`dylibso/observe-sdk`](https://github.com/dylibso/observe-sdk) repository to install the Datadog Adapter for you application.

> In addition, Dylibso provides automatic instrumention tooling which can take any existing WASM module and losslessly recompile it to include function and memory allocation tracing. For

Reviewer 2: ```suggestion
In addition, Dylibso provides automatic instrumentation tooling which can take any existing WASM module and recompile it to include function and memory allocation tracing. For
```

more information, contact [support@dylibso.com](mailto:support@dylibso.com) or learn more about [automatic WebAssembly instrumentation](https://dylibso.com/products/observe).

## Setup

### Installation

Depending on the programming language your application is written in, you should choose one of the appropriate Datadog Adapters from:

>  [`dylibso/observe-sdk`](https://github.com/dylibso/observe-sdk) on GitHub

Reviewer 2: ```suggestion
```

### Configuration

In order to connect the SDK and Adapter to your Datadog Agent, you must have the following information ready:

> 1. Your Datadog Agent host URL

Reviewer 2: ```suggestion
1. Your Datadog Agent host URL.
```

> 2. The service name of the application where the SDK & Adapter are imported

Reviewer 2: ```suggestion
2. The service name of the application where the SDK and Adapter are imported.
```

## Data Collected

### Metrics

WebAssembly Observe SDK can emit standard and custom metrics from your application, depending on your needs.

### Service Checks

WebAssembly Observe SDK does not include any service checks.

### Events

WebAssembly Observe SDK collects logs and traces from your application.

## Troubleshooting

> Need help? Contact [Datadog support][1].

Reviewer: This should be your support email alias

> Once you have imported and configured your Datadog Adapter from the available options within the Observe SDK:

Reviewer 2: ```suggestion
After you have imported and configured your Datadog Adapter from the available options within the Observe SDK:
```

> 2. Ensure that a WebAssembly module (.wasm) has been loaded and you are calling one of its exported functions.

Reviewer 2: ```suggestion
2. Ensure that a WebAssembly module (`.wasm`) has been loaded and you are calling one of its exported functions.
```

# Dylibso WebAssembly Observe SDK

## Overview

This integration provides function traces from WebAssembly (WASM) code running in your application. Gain insight into WebAssembly code performance as well as the following behavior:
- Function call duration
- Execution tracing
- Memory allocation

Since WebAssembly code is executed in a secure and constrained environment, traditional techniques to monitor code do not work. Our specialized observability stack
allows you to continuously monitor WASM modules at the same level you expect of your other applications.

> Datadog customers can use our open source SDKs and Adapters to emit full traces from their WASM programs.

Reviewer 2: ```suggestion
Datadog customers can use our open source SDKs and Adapters to emit full traces from your WASM programs.
```

> Please see the [`dylibso/observe-sdk`][2] repository to install the Datadog Adapter for you application.

Reviewer 2: ```suggestion
Please see the [`dylibso/observe-sdk`][2] repository to install the Datadog Adapter for your application.
```

# Dylibso WebAssembly Observe SDK

## Overview

This integration provides function traces from WebAssembly (WASM) code running in your application. Gain insight into WebAssembly code performance as well as the following behavior:
- Function call duration
- Execution tracing
- Memory allocation

Since WebAssembly code is executed in a secure and constrained environment, traditional techniques to monitor code do not work. Our specialized observability stack
allows you to continuously monitor WASM modules at the same level you expect of your other applications.

Datadog customers can use our open source SDKs and Adapters to emit full traces from your WASM programs.

Please see the [`dylibso/observe-sdk`][2] repository to install the Datadog Adapter for your application.

In addition, Dylibso provides automatic instrumentation tooling which can take any existing WASM module and recompile it to include function and memory allocation tracing. For

> more information, contact [support@dylibso.com](mailto:support@dylibso.com) or learn more about [automatic WebAssembly instrumentation][3].

Reviewer: @estherk15 is this the correct formatting for the email?

Reviewer 2: @emarsha94 Thanks for verifying. Yes, this is correct!

## Setup

### Installation

Depending on the programming language your application is written in, choose one of the appropriate Datadog Adapters from [`dylibso/observe-sdk`][2] on GitHub.

### Configuration

In order to connect the SDK and Adapter to your Datadog Agent, you must have the following information ready:

1. Your Datadog Agent host URL.
2. The service name of the application where the SDK and Adapter are imported.

### Validation

After you have imported and configured your Datadog Adapter from the available options within the Observe SDK:

1. Redeploy your application so the Datadog Adapter is included where you're calling WebAssembly code.
2. Ensure that a WebAssembly module (`.wasm`) has been loaded and you are calling one of its exported functions.
3. Check in your Datadog dashboard for traces sent from your service.

## Data Collected

### Events

> WebAssembly Observe SDK collects logs and traces from your application.

Reviewer: Can we specify that the integration sends traces into datadog?To confirm, you are sending in logs to Datadog? If so, we will need a copy of the raw logs you would send to see if a log pipeline is needed.

Author: Oops I meant to remove logs from here. We do enable users to send their own custom logs from wasm code to Datadog - as such we wouldn't be able to determine the format (up to the application developer). However, we're limiting the focus of this integration tile to be about traces alone for now. 

I'll remove logs from this section, and we can skip the log formatting - does that work?

Reviewer: That works!
